analytics/README.md

```diff
@@ -0,0 +1,54 @@
+# Analytics using Matomo
+
+For this project, we used [matomo](https://matomo.org), a privacy-forward alternative to Google Analytics. See @TODO ADR for reasons why we chose to do so.
+
+## Local development
+
+For local development, there is a `docker-compose.yml` file that will set up a [mysql](https://hub.docker.com/_/mysql) database and an instance of [matomo](https://hub.docker.com/_/matomo).
+
+### First time setup
+
+Matomo does not support automated installation. Instead, a human must manually walk through the browser installer the first time.
+
+1. Start docker
+2. Run `docker compose up -d` and wait for the containers to be ready
+3. Navigate to `localhost` in the browser
+4. Walk through the installation wizard to install the database tables and setup the configuration file (found inside the container at `/var/www/html/config/config.ini.php`)
+5. Stop the docker containers when you are done: `docker compose down`
+
+### Regular usage
+
+After the first time setup is complete, usage is as follows:
+
+- Run `docker compose up -d` to start the database and the matomo container
+- Run `docker compose down` when done
+- To wipe out the database and start over, run `docker compose down -v --remove-orphans`. Walk through the first time setup to start over.
+
+## Deploy to AWS
+
+To deploy to AWS, we use ECS Fargate to host the matomo server, an Aurora mysql database, and an EFS docker volume for persistent files. Deploying to AWS required making some adjustments, including:
+
+- The matomo image by default uses a privileged port (80), which causes issues on AWS ECS Fargate. This is addressed by building a Docker image that uses a `sed` command to change the apache port to 8080, but that can be adjusted with an environment variable.
+- The matomo image is built on the PHP docker image which deploys using Apache. Apache needs to be able to write to `/var/www/html`. This results in a few changes:
+  - Mapping an EFS docker volume to `/var/www/html`
+  - Allowing non-read-only docker root volume
+  - Allow the ECS task to have all EFS IAM permissions (a future @todo would be to refactor this to a more limited scope)
+
+Note: The ECS logs will show errors like: `Operation not permitted: AH02156: setgid: unable to set group id to Group 33`. In addition, the Matomo system check will not return all results until DNS is configured because amazonaws.com is in the blocked hosts list.
```

Comment:
Would prefer a little clarity r.e. `Operation not permitted: AH02156: setgid: unable to set group id to Group 33`.
Is this an expected error because DNS isn't configured, or is it expected because of something else?

Reply:
Good question! I've added this clarity into the README:

> Notes:
> - The ECS logs will show errors like: `Operation not permitted: AH02156: setgid: unable to set group id to Group 33`. This is because Apache will want to try to run SETGID, but because AWS ECS Fargate does not support privileged capabilities such as `SETGID`. See https://docs.aws.amazon.com/AmazonECS/latest/bestpracticesguide/security-tasks-containers.html#security-tasks-containers-recommendations-avoid-privileged-containers for more info. Instead, the EFS has been configured to automatically set posix uid and gid to 33 (e.g. the `www-data` user and group). See infra/modules/file-system/main.tf lines 39-48 and infra/app/env-template/main.tf lines 147-148.
> - The Matomo system check will not return all results until DNS is configured because amazonaws.com is in the blocked hosts list. See https://github.com/matomo-org/matomo/blob/9cb2258632948a3dddc0da92f7d29239a8020b06/config/global.php#L221 This issue should resolve once DNS is configured.